src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -526,5 +590,62 @@ private bool IsV2LocalRepository(string root)
 
             return true;
         }
+
+        // Get a temp API key from nuget.org for pushing to https://nuget.smbsrc.net/
+        private async Task<string> GetSecureApiKey(
+            PackageIdentity packageIdentity,
+            string apiKey,
+            TimeSpan requestTimeout,
+            ILogger logger,
+            CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                return apiKey;
+            }
+            var serviceEndpointUrl = GetServiceEndpointUrl(NuGetConstants.DefaultGalleryServerUrl,
+                string.Format(TempApiKeyServiceEndpoint, packageIdentity.Id, packageIdentity.Version));
+
+            try
+            {
+                var result = await _httpSource.GetJObjectAsync(
+                    new HttpSourceRequest(
+                        () =>
+                        {
+                            var request = HttpRequestMessageFactory.Create(
+                                HttpMethod.Post,
+                                serviceEndpointUrl,
+                                new HttpRequestMessageConfiguration(
+                                    logger: logger,
+                                    promptOn403: false));
+                            request.Headers.Add(ApiKeyHeader, apiKey);
+                            return request;
+                        })
+                    {
+                        RequestTimeout = requestTimeout,
+                        MaxTries = 1
+                    },
+                   logger,
+                   token);
+
+                return result.Value<string>("Key")?? InvalidApiKey;
+            }
+            catch(HttpRequestException ex)
+            {
+                if (ex.Message.Contains("Response status code does not indicate success: 403 (Forbidden)."))
```

Comment:
Can the exception message be localized?  If I install a .NET Framework language pack, will this comparison fail?